<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>KLDebuggerConfigWidgetBase</class>
 <widget class="QDialog" name="KLDebuggerConfigWidgetBase">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>560</width>
    <height>387</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Debugger configuration</string>
  </property>
  <layout class="QGridLayout">
   <item row="1" column="0">
    <widget class="QPushButton" name="pbOK">
     <property name="text">
      <string>&amp;OK</string>
     </property>
     <property name="shortcut">
      <string>Alt+O</string>
     </property>
    </widget>
   </item>
   <item row="1" column="3">
    <widget class="QPushButton" name="pbCancel">
     <property name="text">
      <string>&amp;Cancel</string>
     </property>
     <property name="shortcut">
      <string>Alt+C</string>
     </property>
    </widget>
   </item>
   <item row="1" column="2">
    <spacer name="spacer7">
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Expanding</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>90</width>
       <height>31</height>
      </size>
     </property>
    </spacer>
   </item>
   <item row="1" column="1">
    <widget class="QPushButton" name="pbDefault">
     <property name="text">
      <string>Se&amp;t current settings as default</string>
     </property>
     <property name="shortcut">
      <string>Alt+T</string>
     </property>
    </widget>
   </item>
   <item row="0" column="0" colspan="4">
    <widget class="QTabWidget" name="twSettingsTab">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="TabPage">
      <attribute name="title">
       <string>Common</string>
      </attribute>
      <layout class="QGridLayout">
       <item row="0" column="0">
        <layout class="QHBoxLayout">
         <item>
          <widget class="QLabel" name="tlObjdump">
           <property name="text">
            <string>Objdump command:</string>
           </property>
           <property name="wordWrap">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QLineEdit" name="leObjdump">
           <property name="text">
            <string>avr-objdump</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="1" column="0">
        <spacer name="spacer8">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Expanding</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>91</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>ICD - &amp;Monitor</string>
      </attribute>
      <layout class="QGridLayout">
       <item row="0" column="1">
        <widget class="Q3GroupBox" name="gbInterrupts" native="true">
         <property name="title" stdset="0">
          <string>Interrupts</string>
         </property>
         <layout class="QGridLayout">
          <item row="1" column="0">
           <spacer name="spacer1">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Fixed</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>31</width>
              <height>20</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="0" column="0" colspan="3">
           <widget class="QCheckBox" name="cbCatchInterrupts">
            <property name="text">
             <string>Monitor program catches interrupt re&amp;quests</string>
            </property>
            <property name="shortcut">
             <string>Alt+Q</string>
            </property>
            <property name="checked">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <layout class="QHBoxLayout">
            <item>
             <widget class="QLabel" name="tlInterruptCount">
              <property name="text">
               <string>Interrupt count:</string>
              </property>
              <property name="wordWrap">
               <bool>false</bool>
              </property>
             </widget>
            </item>
            <item>
             <widget class="QSpinBox" name="sbInterruptCount"/>
            </item>
           </layout>
          </item>
          <item row="1" column="2">
           <spacer name="spacer2">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
            <property name="sizeType">
             <enum>QSizePolicy::Expanding</enum>
            </property>
            <property name="sizeHint" stdset="0">
             <size>
              <width>51</width>
              <height>21</height>
             </size>
            </property>
           </spacer>
          </item>
          <item row="2" column="0" colspan="3">
           <widget class="QLabel" name="tlDatasheet">
            <property name="text">
             <string>See the datasheet for details.</string>
            </property>
            <property name="wordWrap">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="Q3GroupBox" name="gbCommunication" native="true">
         <property name="title" stdset="0">
          <string>Communication</string>
         </property>
         <layout class="QGridLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="tlPort">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Fixed" vsizetype="Preferred">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="text">
             <string>Port:</string>
            </property>
            <property name="wordWrap">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QComboBox" name="cbPort">
            <property name="editable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="QLabel" name="tlBAUD">
            <property name="text">
             <string>BAUD:</string>
            </property>
            <property name="wordWrap">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="QComboBox" name="cbBAUD">
            <property name="editable">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="0">
           <widget class="QLabel" name="tlUBRR">
            <property name="text">
             <string>UBRR:</string>
            </property>
            <property name="wordWrap">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="2" column="1">
           <widget class="QSpinBox" name="sbUBRR">
            <property name="minimum">
             <number>0</number>
            </property>
            <property name="maximum">
             <number>255</number>
            </property>
           </widget>
          </item>
          <item row="3" column="0" colspan="2">
           <widget class="QLabel" name="tlError">
            <property name="text">
             <string>Error: &lt;b&gt;0%&lt;/b&gt;</string>
            </property>
            <property name="wordWrap">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="0" colspan="2">
        <widget class="QPushButton" name="pbBuildAndDownload">
         <property name="text">
          <string>Build and upload monitor program &amp;with above settings</string>
         </property>
         <property name="shortcut">
          <string>Alt+W</string>
         </property>
        </widget>
       </item>
       <item row="2" column="0" colspan="2">
        <layout class="QHBoxLayout">
         <item>
          <widget class="QLabel" name="tlCurrentStateLabel">
           <property name="text">
            <string>Current state of the monitor program:</string>
           </property>
           <property name="wordWrap">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <spacer name="spacer5">
           <property name="orientation">
            <enum>Qt::Horizontal</enum>
           </property>
           <property name="sizeType">
            <enum>QSizePolicy::Expanding</enum>
           </property>
           <property name="sizeHint" stdset="0">
            <size>
             <width>21</width>
             <height>20</height>
            </size>
           </property>
          </spacer>
         </item>
         <item>
          <layout class="QVBoxLayout">
           <item>
            <spacer name="spacer3">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeType">
              <enum>QSizePolicy::Expanding</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="KLed" name="kledState" native="true">
             <property name="color" stdset="0">
              <color>
               <red>255</red>
               <green>255</green>
               <blue>0</blue>
              </color>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="spacer4">
             <property name="orientation">
              <enum>Qt::Vertical</enum>
             </property>
             <property name="sizeType">
              <enum>QSizePolicy::Expanding</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>20</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
          </layout>
         </item>
         <item>
          <widget class="QLabel" name="tlState">
           <property name="font">
            <font>
             <weight>75</weight>
             <bold>true</bold>
            </font>
           </property>
           <property name="text">
            <string>unknown</string>
           </property>
           <property name="wordWrap">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="pbCheckState">
           <property name="text">
            <string>Check s&amp;tate</string>
           </property>
           <property name="shortcut">
            <string>Alt+T</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="5" column="0" colspan="2">
        <layout class="QHBoxLayout">
         <item>
          <widget class="QLabel" name="tlMemoryViewByteCount">
           <property name="text">
            <string>Memory view shows first</string>
           </property>
           <property name="wordWrap">
            <bool>false</bool>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QSpinBox" name="sbMemoryViewByteCount">
           <property name="suffix">
            <string> bytes</string>
           </property>
           <property name="maximum">
            <number>65536</number>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item row="4" column="0" colspan="2">
        <widget class="Line" name="line1">
         <property name="frameShape">
          <enum>QFrame::HLine</enum>
         </property>
         <property name="frameShadow">
          <enum>QFrame::Sunken</enum>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <spacer name="spacer6">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Expanding</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>31</height>
          </size>
         </property>
        </spacer>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="TabPage">
      <attribute name="title">
       <string>ICD - Memor&amp;y mapping</string>
      </attribute>
      <layout class="QGridLayout">
       <item row="1" column="0">
        <widget class="QLabel" name="pixmapLabel1">
         <property name="maximumSize">
          <size>
           <width>210</width>
           <height>32767</height>
          </size>
         </property>
         <property name="pixmap">
          <pixmap>image0</pixmap>
         </property>
         <property name="scaledContents">
          <bool>false</bool>
         </property>
         <property name="wordWrap">
          <bool>false</bool>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <spacer name="spacer9">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
         <property name="sizeType">
          <enum>QSizePolicy::Expanding</enum>
         </property>
         <property name="sizeHint" stdset="0">
          <size>
           <width>20</width>
           <height>41</height>
          </size>
         </property>
        </spacer>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="tlWarning">
         <property name="sizePolicy">
          <sizepolicy hsizetype="Minimum" vsizetype="Minimum">
           <horstretch>0</horstretch>
           <verstretch>0</verstretch>
          </sizepolicy>
         </property>
         <property name="maximumSize">
          <size>
           <width>210</width>
           <height>32767</height>
          </size>
         </property>
         <property name="text">
          <string>&lt;b&gt;WARNING:&lt;/b&gt; Wrong settings can cause damages! Use the In Circuit Debugger at &lt;b&gt;YOUR OWN RISK&lt;/b&gt;!
&lt;br&gt;&lt;br&gt;
See the documentation for details!</string>
         </property>
         <property name="textFormat">
          <enum>Qt::AutoText</enum>
         </property>
         <property name="scaledContents">
          <bool>false</bool>
         </property>
         <property name="wordWrap">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="Q3GroupBox" name="groupBox3" native="true">
         <property name="title" stdset="0">
          <string>Define mappings to microcontroller</string>
         </property>
         <layout class="QGridLayout">
          <item row="0" column="0">
           <widget class="QLabel" name="textLabel2">
            <property name="text">
             <string>Map from</string>
            </property>
            <property name="wordWrap">
             <bool>false</bool>
            </property>
           </widget>
          </item>
          <item row="1" column="2">
           <widget class="QPushButton" name="pbAdd">
            <property name="sizePolicy">
             <sizepolicy hsizetype="Preferred" vsizetype="Fixed">
              <horstretch>0</horstretch>
              <verstretch>0</verstretch>
             </sizepolicy>
            </property>
            <property name="maximumSize">
             <size>
              <width>50</width>
              <height>32767</height>
             </size>
            </property>
            <property name="toolTip">
             <string>Add a new mapping</string>
            </property>
            <property name="text">
             <string>Ne&amp;w</string>
            </property>
           </widget>
          </item>
          <item row="1" column="3">
           <widget class="QPushButton" name="pbRemove">
            <property name="maximumSize">
             <size>
              <width>30</width>
              <height>32767</height>
             </size>
            </property>
            <property name="toolTip">
             <string>Removes the selected mapping</string>
            </property>
            <property name="text">
             <string>-</string>
            </property>
           </widget>
          </item>
          <item row="1" column="0">
           <widget class="KIntSpinBox" name="sbFrom" native="true">
            <property name="prefix" stdset="0">
             <string>0x</string>
            </property>
            <property name="maxValue" stdset="0">
             <number>65535</number>
            </property>
            <property name="base" stdset="0">
             <number>16</number>
            </property>
           </widget>
          </item>
          <item row="1" column="1">
           <widget class="KIntSpinBox" name="sbTo" native="true">
            <property name="prefix" stdset="0">
             <string>0x</string>
            </property>
            <property name="maxValue" stdset="0">
             <number>65535</number>
            </property>
            <property name="base" stdset="0">
             <number>16</number>
            </property>
           </widget>
          </item>
          <item row="0" column="1" colspan="2">
           <widget class="QLabel" name="textLabel2_2">
            <property name="text">
             <string>to</string>
            </property>
            <property name="wordWrap">
             <bool>false</bool>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="0" column="1" rowspan="3">
        <widget class="Q3ListView" name="lvMappings" native="true"/>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>Q3GroupBox</class>
   <extends>QWidget</extends>
   <header>q3groupbox.h</header>
  </customwidget>
  <customwidget>
   <class>KLed</class>
   <extends>QWidget</extends>
   <header>kled.h</header>
  </customwidget>
  <customwidget>
   <class>Q3ListView</class>
   <extends>QWidget</extends>
   <header>q3listview.h</header>
  </customwidget>
  <customwidget>
   <class>KIntSpinBox</class>
   <extends>QWidget</extends>
   <header>knuminput.h</header>
  </customwidget>
 </customwidgets>
 <includes>
  <include location="local">kled.h</include>
  <include location="local">knuminput.h</include>
  <include location="local">knuminput.h</include>
 </includes>
 <resources/>
 <connections>
  <connection>
   <sender>pbOK</sender>
   <signal>clicked()</signal>
   <receiver>KLDebuggerConfigWidgetBase</receiver>
   <slot>slotOK()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>29</x>
     <y>371</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pbCancel</sender>
   <signal>clicked()</signal>
   <receiver>KLDebuggerConfigWidgetBase</receiver>
   <slot>slotCancel()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>483</x>
     <y>371</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pbCheckState</sender>
   <signal>clicked()</signal>
   <receiver>KLDebuggerConfigWidgetBase</receiver>
   <slot>slotCheckState()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>459</x>
     <y>217</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pbBuildAndDownload</sender>
   <signal>clicked()</signal>
   <receiver>KLDebuggerConfigWidgetBase</receiver>
   <slot>slotBuildAndDownload()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>31</x>
     <y>168</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>cbBAUD</sender>
   <signal>textChanged(QString)</signal>
   <receiver>KLDebuggerConfigWidgetBase</receiver>
   <slot>slotBAUDRateChanged(QString)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>84</x>
     <y>92</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sbUBRR</sender>
   <signal>valueChanged(int)</signal>
   <receiver>KLDebuggerConfigWidgetBase</receiver>
   <slot>slotUBRRChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>84</x>
     <y>120</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pbDefault</sender>
   <signal>clicked()</signal>
   <receiver>KLDebuggerConfigWidgetBase</receiver>
   <slot>slotSetDefault()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>115</x>
     <y>371</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pbAdd</sender>
   <signal>clicked()</signal>
   <receiver>KLDebuggerConfigWidgetBase</receiver>
   <slot>slotAdd()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>437</x>
     <y>247</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sbFrom</sender>
   <signal>valueChanged(int)</signal>
   <receiver>KLDebuggerConfigWidgetBase</receiver>
   <slot>slotFromChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>245</x>
     <y>247</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>sbTo</sender>
   <signal>valueChanged(int)</signal>
   <receiver>KLDebuggerConfigWidgetBase</receiver>
   <slot>slotToChanged(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>341</x>
     <y>247</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pbRemove</sender>
   <signal>clicked()</signal>
   <receiver>KLDebuggerConfigWidgetBase</receiver>
   <slot>slotRemove()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>493</x>
     <y>247</y>
    </hint>
    <hint type="destinationlabel">
     <x>20</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
